gh-109461: Update logging module lock acquisition to use context manager (#109462)
python/cpython: Misc/NEWS.d/next/Library/2023-09-15-17-12-53.gh-issue-109461.VNFPTK.rst

> logging: Use a context manager for lock acquisition.

Reviewer: ```suggestion
:mod:`logging`: Use a context manager for lock acquisition.
```

> Updated logging library with function "_get_lock" to acquire its module lock using a context manager so that it is safer and easier to use.

Reviewer 2: ```suggestion
logging: Use a context manager for lock acquisition.
```
